lnst/RecipeCommon/Perf/Measurements/LinuxPerfMeasurement.py

```diff
@@ -94,19 +96,24 @@ def start(self) -> None:
         self._start_timestamp = time.time()
         for host, cpu_list_id, cpus in self.configurations:
             filename: str = f"cpu_list{cpu_list_id}.cpus.{'.'.join(map(str, cpus))}.data"
-            self._jobs.append(host.run(LinuxPerf(output_file=filename, cpus=cpus), bg=True))
+            self._running_jobs.append(host.run(LinuxPerf(output_file=filename, cpus=cpus), bg=True))
 
     def finish(self) -> None:
-        for job, (host, _, _) in zip(self._jobs, self.configurations):
+        for job in self._running_jobs:
             job.kill(signal=signal.SIGINT)
+
+        for job, (host, _, _) in zip(self._running_jobs, self.configurations):
             if not job.wait(timeout=120):
                 logging.error(f"timeout while waiting for linuxperf job to finish on host {host.hostid}")
```

Comment:
shouldn't this also do a `job.kill()` with a sigkill to ensure that the background job is stopped in case it is stuck on something and not finishing cleanly?

Reply:
Sure, I'll do that. Just keep in mind that the generated perf.data file won't be valid.

Reply:
Well, if job couldn't save perf file within 120s timeout, I believe something is stuck/broken and we shouldn't use that most likely invalid perf file anyway. Ondrej's point is just to clean up the mess.